bigframes/core/compile/compiler.py

```diff
@@ -268,8 +277,16 @@ def compile_projection(self, node: nodes.ProjectionNode, ordered: bool = True):
     @_compile_node.register
     def compile_concat(self, node: nodes.ConcatNode, ordered: bool = True):
         if ordered:
-            compiled_ordered = [self.compile_ordered_ir(node) for node in node.children]
-            return concat_impl.concat_ordered(compiled_ordered)
+            if self.strict:
+                compiled_ordered = [
+                    self.compile_ordered_ir(node) for node in node.children
+                ]
+                return concat_impl.concat_ordered(compiled_ordered)
+            else:
+                compiled_unordered = [
+                    self.compile_unordered_ir(node) for node in node.children
+                ]
+                return concat_impl.concat_unordered(compiled_unordered).as_ordered_ir()
```

Comment:
This optimization worries me a little more than the join optimization. Could we update some documentation for `concat` to call out this fact? I don't find this intuitive.

Reply:
Hmm, could be confusing if concat doesn't conserve ordering. Reverting this aspect of the change - only join will drop child ordering for now.